packages/integration-testsuite/src/httpSpecTests.ts

```diff
@@ -29,17 +21,18 @@ export function defineIntegrationTestSuiteHttpSpecTests(
         // with GETs. We could override `fetchFn` to add it but this seems simple enough.
         csrfPrevention: false,
       });
-      // Until https://github.com/graphql/graphql-http/issues/24 is addressed,
-      // this hack happens to work.
-      serverAuditOptions.url = createServerResult.url;
+      url = createServerResult.url;
     });
 
     afterAll(async () => {
       await createServerResult?.server.stop();
       await createServerResult?.extraCleanup?.();
     });
 
-    for (const audit of serverAudits(serverAuditOptions)) {
+    for (const audit of serverAudits({
+      url: () => url,
```

Comment:
I think this can just be `createServerResult!.url` and no need for the intermediate `url` variable.

Additionally, maybe `createServerResult` can be defined as:
`let createServerResult: CreateServerForIntegrationTestsResult;`

...eliminating the need for any `!` or `?` as a result of its "possible" nullness.

@glasser is the `| null = null` there for a particular reason?

I suppose we're lying to the type system one way or another.

Reply:
Agreed we don't need a separate variable.
I'm not sure if it works without the `| null = null` but if it does then that's good.

Reply:
Something like this? https://github.com/apollographql/apollo-server/pull/7180/commits/656aeedcb32e7badd0a3d1ced818f352371c6fb2